calyx/src/passes/well_formed.rs

```diff
@@ -217,6 +332,33 @@ impl Visitor for WellFormed {
                 }
             })?;
 
+        if let CellType::Component { name: id } = &cell.prototype {
+            let cellmap = self.external_cell_types.get(id).unwrap();
```

Comment:
You can use indexing syntax when you're certain that a key should exist: `self.external_cell_types[&id]`